incubator/faucet/types/history.go

```diff
@@ -0,0 +1,43 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const (
+	TypeMintHistory = "mint-history"
+)
+
+// NewMining returns a new Mining Message
+func NewMintHistory(minter string, tally int64) *MintHistory {
+	return &MintHistory{
+		Minter:   minter,
```

Comment:
```suggestion
		Minter:   minter.String(),
```

This and the comment above are nits, Its easier to recognize which type of address it should be.

Reply:
still good practice though, will change